components/browser/menu/src/main/java/mozilla/components/browser/menu/WebExtensionBrowserMenuBuilder.kt

```diff
@@ -43,7 +45,24 @@ class WebExtensionBrowserMenuBuilder(
         val extensionMenuItems =
             WebExtensionBrowserMenu.getOrUpdateWebExtensionMenuItems(store.state, store.state.selectedTab)
 
-        val webExtMenuItem = if (extensionMenuItems.isNotEmpty()) {
+        val filteredExtensionMenuItems = extensionMenuItems.toMutableList()
+
+        val finalList = items.toMutableList()
+
+        extensionMenuItems.forEach { webExtensionBrowserMenuItem ->
+            val index = finalList.indexOfFirst { browserMenuItem ->
+                (browserMenuItem as? PlaceholderBrowserMenuItem)?.label ==
+                    (webExtensionBrowserMenuItem as? WebExtensionBrowserMenuItem)?.id
```

Comment:
Let's change `getOrUpdateWebExtensionMenuItems` and `addOrUpdateAction` to operate on a `List<WebExtensionBrowserMenuItem>` instead of a `List<BrowserMenuItem>` as it's really not needed and you can skip the second cast.